site/app/controllers/admin/PlagiarismController.php

```diff
@@ -676,15 +742,20 @@ public function saveNewPlagiarismConfiguration(string $new_or_edit, string $grad
 
         // Generate a unique number for this version of the gradeable //////////
         if ($new_or_edit === "new") {
-            $config_id = 1;
-            if (is_dir(FileUtils::joinPaths($course_path, "lichen", $gradeable_id))) {
-                foreach (scandir(FileUtils::joinPaths($course_path, "lichen", $gradeable_id)) as $file) {
-                    if ($file !== '.' && $file !== '..' && is_numeric($file) && intval($file) >= $config_id) {
-                        $config_id = intval($file) + 1;
-                    }
+            try {
+                $config_id = $em->getRepository(PlagiarismConfig::class)
+                                ->findOneBy(["gradeable_id" => $gradeable_id], ["config_id" => "DESC"]);
+                if ($config_id === null) {
+                    $config_id = 1;
+                }
+                else {
+                    $config_id = $config_id->getConfigID() + 1;
                 }
             }
-            $config_id = strval($config_id);
+            catch (Exception $e) {
```

Comment:
What exceptions are getting throw here? Are these things that'll normally happen, or is it that any exception that gets thrown is because of a bug?